lib/collection.js

```diff
@@ -2999,25 +2664,19 @@ define.classMethod('parallelCollectionScan', { callback: true, promise: true });
  * @return {Promise} returns Promise if no callback passed
  */
 Collection.prototype.geoNear = function(x, y, options, callback) {
-  var self = this;
   var point = typeof x === 'object' && x,
     args = Array.prototype.slice.call(arguments, point ? 1 : 2);
 
   callback = args.pop();
```

Comment:
if you feel like removing 4 more lines of code from each function, I think that just doing

`callback = typeof args[args.length-1] === 'function' ? args.pop() : undefined`

would accomplish the same thing? kinda weird to remove the last element and then sometimes put it back. (I know you didn't introduce that though so feel free to ignore this)

Reply:
oh just realized we already do this in MongoClient.connect lol

Reply:
Good call though! I went through to refactor this and found a number of places I had missed migration to `executeOperation` in GridFS code in the process 💃